packages/adapters/signers/src/eigen.rs

```diff
@@ -0,0 +1,42 @@
+use async_trait::async_trait;
+use rust_eigenda_signers::{Message, PublicKey, RecoverableSignature};
+
+pub mod private_key {
+    pub use rust_eigenda_signers::signers::private_key::Signer;
+}
+pub mod kms;
+
+#[derive(Debug, Clone)]
+pub enum Signer {
+    Private(private_key::Signer),
+    Kms(kms::Signer),
+}
+
+#[async_trait]
+impl eigenda::Sign for Signer {
+    type Error = kms::Error;
+
+    /// Signs a digest using the signer's key.
+    async fn sign_digest(&self, message: &Message) -> Result<RecoverableSignature, Self::Error> {
+        match self {
+            Signer::Private(signer) => {
+                // private_key.sign_digest cannot fail
+                let Ok(sig) = signer.sign_digest(message).await else {
+                    return Err(
+                        anyhow::anyhow!("Failed to sign digest with private key signer").into(),
+                    );
+                };
+                Ok(sig)
+            }
```

Comment:
nit: since `private_key.sign_digest cannot fail`
```suggestion
            Signer::Private(signer) => {
                // private_key.sign_digest cannot fail
                let sig = signer.sign_digest(message).await.expect("Private key signing should never fail");
                Ok(sig)
            }
```

Reply:
addressed in https://github.com/FuelLabs/fuel-block-committer/pull/225